test/scripts/bulk-mailer.js

```diff
@@ -0,0 +1,124 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const ROOT_DIR = '../..'
+
+const { assert } = require('chai')
+const cp = require('child_process')
+const fs = require('fs')
+const mkdirp = require('mkdirp')
+const mocks = require(`${ROOT_DIR}/test/mocks`)
+const path = require('path')
+const P = require('bluebird')
+const rimraf = require('rimraf')
+
+const cwd = path.resolve(__dirname, ROOT_DIR)
+cp.execAsync = P.promisify(cp.exec)
+
+const log = mocks.mockLog()
+const config = require('../../config').getProperties()
+const Token = require('../../lib/tokens')(log, config)
+const UnblockCode = require('../../lib/crypto/random').base32(config.signinUnblock.codeLength)
+
+const OUTPUT_DIRECTORY = path.resolve(__dirname, './test_output')
+const USER_DUMP_PATH = path.join(OUTPUT_DIRECTORY, 'user_dump.json')
+
+describe('scripts/bulk-mailer', () => {
+  let db;
+
+  before(() => {
+    const DB = require('../../lib/db')(
+      config,
+      log,
+      Token,
+      UnblockCode
+    )
+
+    rimraf.sync(OUTPUT_DIRECTORY)
+    mkdirp.sync(OUTPUT_DIRECTORY)
+
+    return DB.connect(config[config.db.backend])
+      .then(_db => {
+        db = _db;
+        return db.createAccount({
+          email: 'user1@test.com',
+          uid: 'f9916686c226415abd06ae550f073cec',
+        })
+      })
+      .then(() => {
+        return db.createAccount({
+          email: 'user2@test.com',
+          uid: 'f9916686c226415abd06ae550f073ced',
+        })
+      })
+      .then(() => {
+        return cp.execAsync(`node scripts/dump-users --uids f9916686c226415abd06ae550f073cec,f9916686c226415abd06ae550f073ced > ${USER_DUMP_PATH}`, { cwd })
+      })
+  });
+
+  after(() => {
+    return db.deleteAccount({
+      uid: 'f9916686c226415abd06ae550f073cec'
+    }).then(() => {
+      return db.deleteAccount({
+        uid: 'f9916686c226415abd06ae550f073ced'
+      })
+    }).then(() => {
+      return db.close()
+    }).then(() => {
+      return rimraf.sync(OUTPUT_DIRECTORY)
+    })
+  })
+
+  it('fails if --input missing', () => {
+    return cp.execAsync('node scripts/bulk-mailer --method sendVerifyCode', { cwd })
+      .then(() => assert(false, 'script should have failed'), () => {})
```

Comment:
I think we might be able to check the output and assert correct error message for some of these.